### DISCUSSION TITLE
【小白】请教一下为啥我生成的 ERC20 钱包地址和老师生成的长度不一样呢？

### DISCUSSION THREAD

**Discussion by hu-qi** [openbuildxyz/Web3-Frontend-Bootcamp#132] [Q&A] [answered]
<img width="725" alt="image" src="https://github.com/openbuildxyz/Web3-Frontend-Bootcamp/assets/17986122/db517a30-3368-4ffa-bbf4-5f29265dfc02">

<img width="652" alt="image" src="https://github.com/openbuildxyz/Web3-Frontend-Bootcamp/assets/17986122/e1081646-f309-46d6-baed-f866a6131fd3">

老师能解答一下这个问题吗？涉及到哪些知识点？感谢！

**Comment by anderfib** [comment, marked as answer]
your address is right!
以太系列的地址生成 用  0x + 哈希值的后40位得到 所以地址长度应该是42位
至于这个哈希值怎么得到 你可以问问ai

**Comment by LenisLee** [comment]
以你自己的钱包为准，demo只是举个例子

**Comment by gamble369** [comment]
可以试试，将示例中的钱包地址填入私钥，具体位置【dd account or hardware wallet】->【导入账户】->【选择类型：私钥】
可以得到一个新账号(只不过 no money )；

**Comment by linghuccc** [comment]
因为老师写错了，你的才是正确的钱包地址，老师给的是私钥，是错误的示范
